cli/internal/run/real_run.go

```diff
@@ -72,6 +72,11 @@ func RealRun(
 
 	runCache := runcache.New(turboCache, base.RepoRoot, rs.Opts.runcacheOpts, colorCache)
 
+	globalEnvMode := rs.Opts.runOpts.EnvMode
+	if globalEnvMode == util.Infer && turboJSON.GlobalPassthroughEnv != nil {
+		globalEnvMode = util.Strict
+	}
```

Comment:
I'd really like not to have any duplication between run and dry run (Took great pains to remove all the duplications that we had before). Is it possible to avoid thiS?

Reply:
This can be avoided by hoisting the calculation earlier.

Aside: the way in which those were all collapsed into `GlobalHashable` is creating lots of pain because it now tightly couples real_run, dry_run, globalhash, and runsummary. Particularly, `envVars` is in an "interim state" and the rest are "resolved state". `envVars` is an `env.DetailedMap` and presently getting hashed as is when it should instead be `.All.ToHashable()`.

Reply:
Hoisted to `run`.